src/player/Player.php

```diff
@@ -1814,7 +1814,10 @@ public function interactEntity(Entity $entity, Vector3 $clickPos) : bool{
 
 		$ev->call();
 
+		$item = $this->inventory->getItemInHand();
 		if(!$ev->isCancelled()){
+			$item->onInteractEntity($this, $entity, $clickPos);
```

Comment:
This isn't enough by itself. You need to check if the item was changed, and set it back into the inventory if it was.

For example, using shears will change the durability, and using a nametag will reduce the count by 1.